sdk/identity/identity/src/credentials/workloadIdentityCredential.ts

```diff
@@ -56,21 +62,66 @@ export class WorkloadIdentityCredential implements TokenCredential {
     const workloadIdentityCredentialOptions = options ?? {};
     const tenantId = workloadIdentityCredentialOptions.tenantId || process.env.AZURE_TENANT_ID;
     const clientId = workloadIdentityCredentialOptions.clientId || process.env.AZURE_CLIENT_ID;
+
     this.federatedTokenFilePath =
-      workloadIdentityCredentialOptions.tokenFilePath || process.env.AZURE_FEDERATED_TOKEN_FILE;
+      workloadIdentityCredentialOptions?.tokenFilePath || process.env.AZURE_FEDERATED_TOKEN_FILE;
+    this.serviceConnectionId = workloadIdentityCredentialOptions.serviceConnectionId;
+    if (!tenantId || !clientId) {
+      throw new CredentialUnavailableError(
+        `${credentialName}: is Unavailable. clientId and tenantId are required parameters/ environment varables - AZURE_TENANT_ID, AZURE_CLIENT_ID`
+      );
+    }
     if (tenantId) {
       checkTenantId(logger, tenantId);
     }
+    if (this.federatedTokenFilePath && this.serviceConnectionId) {
+      const errorMessage = `${credentialName}: ambiguous. serviceConnectionId and tokenFilePath cannot be provided at the same time. These are used for supporting Workload Identity for different environments. 
+      The tokenFilePath is used for Kubernetes while serviceConnectionId is used for Azure Pipelines.
+      See the troubleshooting guide for more information: https://aka.ms/azsdk/js/identity/workloadidentitycredential/troubleshoot`;
+      logger.error(errorMessage);
+      throw new CredentialUnavailableError(errorMessage);
+    }
+
     if (clientId && tenantId && this.federatedTokenFilePath) {
```

Comment:
clientId and tenantId should always be set by now, you can remove check for them